astropy/io/fits/hdu/table.py

```diff
@@ -1138,15 +1139,15 @@ def _dump_data(self, fileobj):
         def format_value(val, format):
             if format[0] == 'S':
                 itemsize = int(format[1:])
-                return '%-*s' % (itemsize, val)
+                return '{:{size}}'.format(val, size=itemsize)
             elif format in np.typecodes['AllInteger']:
                 # output integer
-                return '%21d' % val
+                return '{:21d}'.format(val)
             elif format in np.typecodes['Complex']:
-                return '%21.15g+%.15gj' % (val.real, val.imag)
+                return '{:21.15g}+{:.15g}j'.format(val.real, val.imag)
             elif format in np.typecodes['Float']:
                 # output floating point
-                return '%#21.15g' % val
+                return '{:21.15g}'.format(val)
```

Comment:
Yes. atm both versions do the same by removing the trailing zeros. But since this is the fits module, I can never see whether it's important to keep them for an obscure case or not, so I would slightly prefer to keep the original behaviour at least for python3.